### ISSUE TITLE
Add CKAN to support Solr 5.0

### ISSUE THREAD

**Issue by winggundamth** [ckan/ckan#2318]
Since Solr 5.0 has been released http://lucene.apache.org/solr/news.html. I tried to test it with CKAN latest version but when I'm trying to reindex but I found this error

```
root@ckan:/# paster --plugin=ckan search-index rebuild --config=/etc/ckan/production.ini
2015-02-24 16:32:33,520 ERROR [ckan.lib.search.common] HTTP code=404, reason=Not Found
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/ckan/lib/search/common.py", line 51, in is_available
    conn.query("*:*", rows=1)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 703, in query
    return self.select(*args, **params)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 798, in __call__
    xml = self.raw(**params)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 823, in raw
    rsp = conn._post(self.selector, request, conn.form_headers)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 639, in _post
    return check_response_status(self.conn.getresponse())
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 1097, in check_response_status
    raise ex
SolrException: HTTP code=404, reason=Not Found
2015-02-24 16:32:33,521 WARNI [ckan.lib.search] Problems were found while connecting to the SOLR server
2015-02-24 16:32:33,703 ERROR [ckan.lib.search.common] HTTP code=404, reason=Not Found
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/ckan/lib/search/common.py", line 51, in is_available
    conn.query("*:*", rows=1)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 703, in query
    return self.select(*args, **params)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 798, in __call__
    xml = self.raw(**params)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 823, in raw
    rsp = conn._post(self.selector, request, conn.form_headers)
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 639, in _post
    return check_response_status(self.conn.getresponse())
  File "/usr/local/lib/python2.7/dist-packages/solr/core.py", line 1097, in check_response_status
    raise ex
SolrException: HTTP code=404, reason=Not Found
```

Reference from here https://issues.apache.org/jira/browse/SOLR-6852 I see that Solr 5.0 has been changed update url from http://solr.example.com/update to http://solr.example.com/collection1/update. So I'm not sure how to config CKAN or it's a new feature that need to add to support Solr 5.0

**Comment by joetsoi** [comment]
At the very least a bunch of our docs need to change.

https://cwiki.apache.org/confluence/display/solr/Major+Changes+from+Solr+4+to+Solr+5
https://wiki.apache.org/solr/Solr.xml%204.4%20and%20beyond

**Comment by joetsoi** [comment]
@winggundamth you need to change your solr_url in your development.ini to
`solr_url = http://solr.example.com/collection1/update`

I tested solr 5.0 and it is running fine.

we need to change the docs, but I don't see solr 5.0 being packaged for ubuntu 12.10/14.04 anytime soon, but when it is, we'll need to change the docs then.

**Comment by ykhadilkar-rei** [comment]
I tried updating solr_url to http://solr.example.com/collection1/update Still getting similar error.

**Comment by joetsoi** [comment]
hey you should probably just change it to `http://<your solr server>/collection1` or change collection1 to the core name that you setup.

**Comment by ykhadilkar-rei** [comment]
I did. Above url is just for format/ reference purpose. Switched to 4.9 works on that.

**Comment by joetsoi** [comment]
my recommendation is to go to `http://<solr>/solr` select your core in the drop down menu on the left, and then click on "Query", then click "Execute Query" you should see a query being executed and the correct url to use see  https://lucene.apache.org/solr/assets/images/quickstart-query-screen.png

**Comment by Dobedani** [comment]
Sorry to break into the technical discussion. I'm relatively new to CKAN. I've tried to get version 2.2 running without problems. My aim is to develop extensions and at this moment I can only do so on Windows, so I tried to get CKAN 2.2 running on Windows - succeeded somehow. I'd like to express that even on Linux I experienced Solr as a major stumbling block in the process of getting CKAN to work. Now I understand that I should have tried a lower version than 5.1.0. The mirrors provide version 4.7.2 as the oldest version. And now I understand that I should have created a core. Well, there seems to be a lot of documentation on how to administer Solr, but I could hardly see the wood for the trees. E.g. only after much trial and error I found out how to create a core - c.q. with name ckan2.2 - and to get a response other than 404 upon this simple request: http://localhost:8983/solr/ckan2.2/select/?q=*:*&wt=json
If the documentation can be improved, I think it would help a lot. Thanks in advance!
